modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxBinaryOption.java

```diff
@@ -0,0 +1,856 @@
+/**
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.fxopt;
+
+import static com.opengamma.strata.collect.ArgChecker.inOrderOrEqual;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import com.opengamma.strata.basics.currency.AdjustablePayment;
+import com.opengamma.strata.basics.index.FxIndex;
+import com.opengamma.strata.product.common.PutCall;
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableValidator;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.Resolvable;
+import com.opengamma.strata.product.Product;
+import com.opengamma.strata.product.common.LongShort;
+import com.opengamma.strata.basics.currency.CurrencyPair;
+
+/**
+ * A binary FX option.
+ * <p>
+ * A binary FX option is a financial instrument that provides a payoff based on the future value of
+ * a foreign exchange index. The currency unit of the payoff can be in either currency unit of the foreign exchange.
+ * The option is European, exercised only on the exercise date.
+ * <p>
+ * For example, a binary call with a strike rate 'EUR 1.00 / USD 1.1' and a payment amount USD 100 pays
+ * this amount at expiry if the future value of the foreign exchange index is at least as great as the
+ * strike rate.
+ */
+@BeanDefinition
+public final class FxBinaryOption
+        implements Product, Resolvable<ResolvedFxBinaryOption>, ImmutableBean, Serializable {
+
+    /**
```

Comment:
OpenGamma uses indentation at 2 characters, not 4. Need to change your settings and reformat classes.